internal/appsec/dyngo/instrumentation/httpsec/http.go

```diff
@@ -54,16 +50,26 @@ func WrapHandler(handler http.Handler, span ddtrace.Span) http.Handler {
 	span.SetTag("_dd.runtime_family", "go")
 
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		args := MakeHandlerOperationArgs(r)
 		op := StartOperation(
-			makeHandlerOperationArgs(r, span),
+			args,
 			nil,
 		)
 		defer func() {
 			var status int
 			if mw, ok := w.(interface{ Status() int }); ok {
 				status = mw.Status()
 			}
-			op.Finish(HandlerOperationRes{Status: status})
+			events := op.Finish(HandlerOperationRes{Status: status})
+			if len(events) == 0 {
+				return
+			}
+
+			remoteIP, _, err := net.SplitHostPort(r.RemoteAddr)
+			if err != nil {
+				remoteIP = r.RemoteAddr
+			}
+			SetSecurityEventTags(span, events, remoteIP, args.Headers)
```

Comment:
Improvement: the tags are set in the integration package which has access to everything we need to set the span tag, without having to pass them into the HandlerOperationArgs anymore.